api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,172 @@
+  {
+    Context context      = Context(key, value);
+    context.head_->next_ = head_;
+    return context;
+  }
+
+  // Returns the value associated with the passed in key.
+  context::ContextValue GetValue(nostd::string_view key)
+  {
+    for (DataList *data = &*head_; data != nullptr; data = &*(data->next_))
+    {
+      if (key.size() == data->key_length_)
+      {
+        if (memcmp(key.data(), data->key_, data->key_length_) == 0)
+        {
+          return data->value_;
+        }
+      }
+    }
+    return (int64_t)0;
+  }
+
+  // Checks for key and returns true if found
+  bool HasKey(nostd::string_view key)
+  {
+    for (nostd::shared_ptr<DataList> data = head_; data != nullptr; data = data->next_)
+    {
+      if (key.size() == data->key_length_)
+      {
+        if (memcmp(key.data(), data->key_, data->key_length_) == 0)
+        {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+private:
+  Context() = default;
+
+  // A linked list to contain the keys and values of this context node
+  class DataList
+  {
+  public:
+    nostd::shared_ptr<DataList> next_;
+
+    char *key_;
+
+    size_t key_length_;
+
+    ContextValue value_;
+
+    DataList() { next_ = nullptr; }
+
+    // Builds a data list off of a key and value iterable and returns the head
+    template <class T>
+    DataList(const T &keys_and_vals)
+    {
+      if (std::begin(keys_and_vals) != std::end(keys_and_vals))
+      {
+        auto iter = std::begin(keys_and_vals);
+        // Create list head
+        key_        = new char[nostd::string_view(iter->first).size()];
+        key_length_ = nostd::string_view(iter->first).size();
+        memcpy(key_, nostd::string_view(iter->first).data(),
+               nostd::string_view(iter->first).size() * sizeof(char));
+        value_ = iter->second;
+        next_  = nostd::shared_ptr<DataList>{nullptr};
+        ++iter;
+
+        DataList *previous_node = this;
+        // Iterate over the keys and values iterable and add nodes
+        for (; iter != std::end(keys_and_vals); ++iter)
+        {
+          nostd::shared_ptr<DataList> node = nostd::shared_ptr<DataList>{new DataList()};
+          node->key_                       = new char[nostd::string_view(iter->first).size()];
+          node->key_length_                = nostd::string_view(iter->first).size();
+          memcpy(node->key_, nostd::string_view(iter->first).data(),
+                 nostd::string_view(iter->first).size() * sizeof(char));
+          node->value_         = iter->second;
+          previous_node->next_ = node;
+          previous_node        = &*node;
+        }
+      }
+      else
+      {
+        key_ = nullptr;
+      }
+    }
+
+    // Builds a data list with just a key and value, so it will just be the head
+    // and returns that head.
+    DataList(nostd::string_view key, ContextValue value)
+    {
+      key_        = new char[nostd::string_view(key).size()];
```

Comment:
No need for the explicit `nostd::string_view` here, this can just be `key.size()`.

Reply:
Removed.